src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2045,31 +2051,40 @@ int wm_vuldet_linux_oval_vulnerabilities(sqlite3 *db, agent_software *agents_it,
 
     wdb_finalize(stmt);
 
-    mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_VULN_OVAL_AG_COUNT, vuln_count, atoi(agents_it->agent_id));
-    mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_FUNCTION_TIME, time(NULL) - start_time, "find OVAL", atoi(agents_it->agent_id));
+    mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_VULN_OVAL_AG_COUNT, vuln_count, scan_ctx->agent_id);
+    mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_FUNCTION_TIME, time(NULL) - start_time, "find OVAL", scan_ctx->agent_id);
 
     return 0;
 }
 
-int wm_vuldet_report_agent_vulnerabilities(sqlite3 *db, agent_software *agent, wm_vuldet_flags *flags) {
+int wm_vuldet_find_agent_vulnerabilities(sqlite3 *db, scan_agent *agent, wm_vuldet_flags *flags, scan_ctx_t* scan_ctx) {
 
     int retval = OS_INVALID;
     OSHash *cve_table = NULL;
+    int sock = wm_vuldet_get_wdb_socket();
 
-    if (!agent->info) {
+    if (!scan_ctx->os_scan && !scan_ctx->package_scan) {
         retval = 0;   // Scan next agent
         goto end;
     }
 
-    if (agent->dist == FEED_WIN) {
+    //Set the pending status previous to the scan
+    if (scan_ctx->scan_type == VU_FULL_SCAN) {
+        wdb_agents_vuln_cves_update_status(scan_ctx->agent_id, "*", VULN_CVES_STATUS_PENDING, &sock);
+    }
+    else if(scan_ctx->os_scan) {
```

Comment:
As per the discussion that we had, probably we should also check here if the scan is a partial scan.

Reply:
Fixed